dask/dataframe/utils.py

```
@@ -237,7 +238,21 @@ def _empty_series(name, dtype, index=None):
     return pd.Series([], dtype=dtype, name=name, index=index)
 
 
-def make_meta(x, index=None):
+make_meta = Dispatch('make_meta')
```

This is a public method (it's referred to in many docstrings, as well as the docs). I'd rather not change the signature of this method - can the dispatch method be named something else, and `make_meta` calls the dispatch internally where needed?

---

The new dispatch function is *almost* the same.  I suspect that it would be exactly the same if we added the following implementation

```python
@make_meta.register(object, object)
def make_meta_two_args(a, index):
    return make_meta(a, index=index)
```

Are there other corner cases?

---

That doesn't work, `Dispatch` is for single dispatch, not multipledispatch. It's designed only to dispatch and propagate a single argument through multiple implementations of a function.

My concerns are as follows:
- The signature change
- Lack of a docstring on the `make_meta` object (where there previously was one).

---

Ah, you're right (although please do take a look at the changes to Dispatch in this PR.  They don't affect this issue, but it would be good to get your thoughts on them).

I think that I could partially resolve those concerns by ...

1.  Including an `index=` keyword in other places (though we couldn't enforce this)
2.  Improve docstring handling on `Dispatch` objects

---

I've now changed Dispatch to take the docstring from the `object` method

Index works everywhere it used to.  However now the implementations for DataFrame and Series will err of the user passes in an index.  Previously they would silently ignore it.

---

Seems good to me.